pints/_log_priors.py

```diff
@@ -427,14 +432,145 @@ def sample(self, n=1):
                                           scale=self._scale, size=n)
 
 
+class InverseGammaLogPrior(pints.LogPrior):
+    """
+    Defines an inverse gamma (log) prior with given shape parameter ``a`` and
+    scale parameter ``b``, with pdf
+
+    .. math::
+        f(x|a,b)=\\begin{cases}\\frac{b^a}{\\Gamma(a)}x^{-a-1}\\exp
+        \\left(-\\frac{b}{x}\\right),&x>0\\\\0,&\\text{Otherwise}\\end{cases}
+
+    where :math:`\\Gamma` is the Gamma function.  This pdf has expectation
+
+    .. math::
+        \\mathrm{E}(X)=\\begin{cases}\\frac{b}{a-1},&a>1\\\\
+        \\text{undefined},&\\text{otherwise.}\\end{cases}
+
+    For example, to create a prior with shape parameter ``a=5`` and scale
+    parameter ``b=1``, use::
+
+        p = pints.InverseGammaLogPrior(5, 1)
+
+    *Extends:* :class:`LogPrior`
+    """
+    def __init__(self, a, b):
+        # Parse input arguments
+        self._a = float(a)
+        self._b = float(b)
+
+        # Validate inputs
+        if self._a <= 0:
+            raise ValueError('Shape parameter a must be positive')
+        if self._b <= 0:
+            raise ValueError('Scale parameter b must be positive')
+
+        # Cache constants
+        self._k = self._a * np.log(self._b) - scipy.special.gammaln(self._a)
+        self._ap1 = self._a + 1.
+
+    def __call__(self, x):
+        _x = float(x[0])
+
+        if _x <= 0.0:
+            return -np.inf
+        else:
+            return self._k - self._ap1 * np.log(_x) - np.divide(self._b, _x)
+
+    def evaluateS1(self, x):
+        """ See :meth:`LogPDF.evaluateS1()`. """
+        val = self(x)
+
+        _x = float(x[0])
+
+        if _x < 0.0:
+            return val, np.asarray([0.])
+        else:
+            return val, np.asarray(
+                [np.divide(self._b - self._ap1 * _x, _x * _x)])
+
+    def n_parameters(self):
+        """ See :meth:`LogPrior.n_parameters()`. """
+        return 1
+
+    def sample(self, n=1):
+        """ See :meth:`LogPrior.sample()`. """
+        return scipy.stats.invgamma.rvs(a=self._a, scale=self._b, loc=0.,
+                                        size=n)
+
+
+class LogNormalLogPrior(pints.LogPrior):
+    r"""
+    Defines a log-normal (log) prior with a given ``logmean`` and scale
+    ``scale``, with pdf
+
+    .. math::
+        f(x|\text{logmean},\text{scale}) = \frac{1}{x\;\text{scale}\sqrt{2\pi}}
+        \exp\left(-\frac{(\log x-\text{logmean})^2}{2\;\text{scale}^2}\right)
+
+    and expectation
```

Comment:
I think it's worth creating mean functions for the log priors. I will make this a separate ticket. This makes the job of a modeller even easier since they just call this function to check the parameterisation...